File: Aula_12/banco.js
class ContaBancaria {
    constructor(numero, titular) {
        this.numero = numero;
        this.titular = titular;
        this.saldo = 0.0;
        this.historico = [];
    }
    depositar(valor) {
        if (valor > 0) {
            this.saldo += valor;
            this.historico.push(`Depósito: R$ ${valor.toFixed(2)}`);
            console.log(`Valor depositado com sucesso! Seu saldo é de R$ ${this.saldo.toFixed(2)}`)
        } else {
            console.log(`Sinto muito!Valor inválido para deposito.`)
        }
    }
    sacar(valor) {
        if (valor > 0 && this.saldo >= valor) {
            this.saldo -= valor;
            this.historico.push(`Saque: R$ ${valor.toFixed(2)}`);
            console.log(`Valor sacado com sucesso! Seu saldo é de R$ ${this.saldo.toFixed(2)}`)
        } else {
            console.log(`Sinto muito!Saldo insuficiente ou Valor inválido para saque.`)
        }
    }
    mostraStatus() {
        console.log(`\n-------CONTA ${this.numero.toUpperCase()}`);
        console.log(`Titular da conta: ${this.titular}`);
        console.log(`Saldo na conta: ${this.saldo.toFixed()}`);
    }

    mostrar_extrato() {
        console.log(`------EXTRATO CONSOLIDADO DA CONTA ${this.numero}------`)
        console.log(`${this.titular}`)
        console.log(`Seu saldo é de R$ ${this.saldo}`)
        console.log(`\nÚltimas transações:`)

        let UltimasTransacoes = this.historico.slice(-5);
        if (UltimasTransacoes.length > 0) {
            for (let i = 0; i < UltimasTransacoes.length; i++) {
                console.log(`${i + 1}: ${UltimasTransacoes[i]}`)
            }
        } else {
            console.log(`Nenhuma transação efetivada até o momento!`)
        }
    }
}

// Programa principal

let conta = new ContaBancaria("001", "João Silva");
let rodando = true;
while (rodando) {
    console.log(`\n===BANCO JWC===`);
    console.log(`1 - Depositar`);
    console.log(`2 - Sacar`);
    console.log(`3 - Extrato`);
    console.log(`4 - Status da Conta`);
    console.log(`0 - Sair`);

    let opcao = prompt(` Escolha uma opção: `);
    if (opcao === "1") {
        let valorDeposito = parseFloat(prompt(`Valor do depósito é de R$`));
        if (!isNaN(valorDeposito)) {
            conta.depositar(valorDeposito);
        } else {
            console.log("Digite um número válido");
        }
        prompt(`Pressiona Enter para continuar...`);
    } else if (opcao === "2") {
        let valorSaque = parseFloat(prompt(`Valor do saque é de R$`));
        if (!isNaN(valorSaque)) {
            conta.sacar(valorSaque)
        } else {
            console.log("Digite um número válido");
        }
        prompt(`Pressiona Enter para continuar...`);
    } else if (opcao === "3") {
        console.log(conta.mostrar_extrato());
        prompt(`Pressiona Enter para continuar...`);
    } else if (opcao === "4") {
        console.log(conta.mostraStatus());
        prompt(`Pressiona Enter para continuar...`);
    } else if (opcao === "0") {
        console.log(`Obrigado por usar o banco JWC. Volte sempre!`);
        rodando = false;
    } else {
        console.log(`Opção inválida! Tente novamente.`)
        prompt(`Pressiona Enter para continuar...`);
    }

}

File: Aula_12/banco.py
class ContaBancaria:
    def __init__(self, numero, titular):
        self.numero = numero
        self.titular = titular
        self.saldo = 0.0
        self.historico = []
    
    def depositar(self, valor):
        if valor > 0:
            self.saldo += valor
            self.historico.append(f"Depósito: +R${valor:.2f}")
            print(f"Depósito de R${valor:.2f} realizado. Saldo: R${self.saldo:.2f}")
        else:
            print("Valor inválido para depósito.")
    
    def sacar(self, valor):
        if valor > 0 and self.saldo >= valor:
            self.saldo -= valor
            self.historico.append(f"Saque: -R${valor:.2f}")
            print(f"Saque de R${valor:.2f} realizado. Saldo: R${self.saldo:.2f}")
        else:
            print("Saldo insuficiente ou valor inválido.")
    
    def mostrar_status(self):
        print(f"\n=== CONTA {self.numero.upper()} ===")
        print(f"Titular: {self.titular}")
        print(f"Saldo atual: R${self.saldo:.2f}")
    
    def mostrar_extrato(self):
        print(f"\n=== EXTRATO CONTA {self.numero} ===")
        print(f"Titular: {self.titular}")
        print(f"Saldo: R${self.saldo:.2f}")
        print("\nÚltimas transações:")
        
        # Laço for para últimas 5 transações
        ultimas = self.historico[-5:]  # Pega últimas 5
        if ultimas:
            for i, transacao in enumerate(ultimas, 1):
                print(f"{i}. {transacao}")
        else:
            print("Nenhuma transação ainda.")
        print("-" * 40)

# === PROGRAMA PRINCIPAL ===
conta = ContaBancaria("001", "João Silva")

while True:
    print("\n=== BANCO JWC ===")
    print("1 - Depositar")
    print("2 - Sacar")
    print("3 - Extrato")
    print("4 - Status")
    print("0 - Sair")
    
    opcao = input("Escolha uma opção: ").strip()
    
    if opcao == "1":
        try:
            valor = float(input("Valor do depósito: R$"))
            conta.depositar(valor)
        except ValueError:
            print("Digite um número válido.")
    
    elif opcao == "2":
        try:
            valor = float(input("Valor do saque: R$"))
            conta.sacar(valor)
        except ValueError:
            print("Digite um número válido.")
    
    elif opcao == "3":
        conta.mostrar_extrato()
    
    elif opcao == "4":
        conta.mostrar_status()
    
    elif opcao == "0":
        print("Obrigado por usar o Banco JWC!")
        break
    
    else:
        print("Opção inválida. Tente novamente.")
